analytics_data_api/utils.py

```diff
@@ -84,3 +94,92 @@ def date_range(start_date, end_date, delta=datetime.timedelta(days=1)):
     while cur_date < end_date:
         yield cur_date
         cur_date += delta
+
+
+def get_course_report_download_details(course_id, report_name):
+    """
+    Determine the path that the report file should be located at,
+    then return metadata sufficient for downloading it.
+    """
+    report_location_template = getattr(settings, 'COURSE_REPORT_FILE_LOCATION_TEMPLATE')
+    report_location = report_location_template.format(
+        course_id=course_id,
+        report_name=report_name
+    )
+    if getattr(settings, 'DEFAULT_FILE_STORAGE', None) != AWS_S3_FILE_STORAGE_CLASS:
```

Comment:
Ah, I got thrown by [files.storage.Storage.url()](https://docs.djangoproject.com/en/1.9/ref/files/storage/#django.core.files.storage.Storage.url).

Yes, could you please try/catch that one, and throw your custom "CannotCreateReportDownloadLinkError" exception on TypeError or NotImplementedError?